go/vt/vtgate/engine/comparer.go

```diff
@@ -23,16 +23,22 @@ import (
 
 // comparer is the struct that has the logic for comparing two rows in the result set
 type comparer struct {
-	orderBy, weightString int
-	desc                  bool
+	orderBy, weightString, starColFixedIndex int
+	desc                                     bool
 }
 
 // compare compares two rows given the comparer and returns which one should be earlier in the result set
 // -1 if the first row should be earlier
 // 1 is the second row should be earlier
 // 0 if both the rows have equal ordering
 func (c *comparer) compare(r1, r2 []sqltypes.Value) (int, error) {
-	cmp, err := evalengine.NullsafeCompare(r1[c.orderBy], r2[c.orderBy])
+	var colIndex int
+	if c.starColFixedIndex > c.orderBy && c.starColFixedIndex < len(r1) {
```

Comment:
this will be the same for every comparison, so it would be preferable to not have to calculate the `colIndex` for every comparison

Reply:
I have no idea to avoid this comparison, unless I define another attribute in comparer to save the finally compare var.
Do you have any suggests?

Reply:
Yeah, I see what you mean. This will have to do for now